client/checkout/utils/copy-test-number.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+document.addEventListener(
+	'click',
+	function ( event ) {
+		const copyNumberButton = event.target?.closest(
+			'.js-woopayments-copy-test-number'
+		);
+		if ( copyNumberButton ) {
+			event.preventDefault();
+			const number = copyNumberButton.parentElement.querySelector(
+				'.js-woopayments-test-number'
+			).innerText;
+			navigator.clipboard.writeText( number );
+
+			window.wp?.data
```

Comment:
`window.wp.data` might not be present on shortcode checkout, hence the optional chaining